controllers/cloud.redhat.com/clowdenvironment_controller.go

```diff
@@ -75,30 +75,35 @@ func (r *ClowdenvironmentReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{Requeue: true}, fmt.Errorf("error setting annotations: %w", err)
 	}
 
-	ns, err := helpers.GetNamespace(ctx, r.Client, nsName)
+	namespace, err := helpers.GetNamespace(ctx, r.Client, nsName)
 	if err != nil {
-		r.Log.Error(err, "could not retrieve newly created namespace", "namespace", nsName)
+		r.Log.Error(err, "could not retrieve updated resource", "namespace", nsName)
 	}
 
-	if _, ok := ns.Annotations[helpers.CompletionTime]; ok {
+	if _, ok := namespace.Annotations[helpers.CompletionTime]; ok {
 		return ctrl.Result{}, nil
 	}
 
 	nsCompletionTime := time.Now()
 	var AnnotationCompletionTime = helpers.CustomAnnotation{Annotation: helpers.CompletionTime, Value: nsCompletionTime.String()}
 
-	err = helpers.UpdateAnnotations(ctx, r.Client, ns.Name, AnnotationCompletionTime.ToMap())
+	err = helpers.UpdateAnnotations(ctx, r.Client, namespace.Name, AnnotationCompletionTime.ToMap())
 	if err != nil {
 		r.Log.Error(err, "could not update annotation with completion time", "namespace", nsName)
 	}
 
-	if err := r.Client.Update(ctx, &ns); err != nil {
-		return ctrl.Result{}, err
+	namespace, err = helpers.GetNamespace(ctx, r.Client, nsName)
+	if err != nil {
+		r.Log.Error(err, "could not retrieve newly created namespace", "namespace", nsName)
```

Comment:
Yeah same question here too - is logging the error here enough, or should something else happen? Also, this looks like a straight copypasta of line 78 -that *may* be ok - but in my own code when I see copypasta a dozen lines a part in the same method I start to wonder if I could tighten things up. Maybe not - this is Golang where sometimes you just gotta do stuff like this - but throwing the idea out there for your consideration